rasterio/_warp.pyx

```diff
@@ -169,25 +174,72 @@ def _reproject(
     If the source and destination are rasterio Bands, shorthand for
     bands of datasets on disk, the coordinate reference systems and
     transforms will be read from the appropriate datasets.
+
+    Parameters
+    ------------
+    source: ndarray or rasterio Band
+        Source raster.
+    destination: ndarray or rasterio Band
+        Target raster.
+    src_transform: affine transform object, optional
+        Source affine transformation.  Required if source and destination
+        are ndarrays.  Will be derived from source if it is a rasterio Band.
+    src_crs: dict, optional
+        Source coordinate reference system, in rasterio dict format.
+        Required if source and destination are ndarrays.
+        Will be derived from source if it is a rasterio Band.
+        Example: {'init': 'EPSG:4326'}
+    src_nodata: int or float, optional
+        The source nodata value.  If not set, it will be default to the
+        nodata value of the source image if a masked ndarray or rasterio band,
+        if available.  Must be provided if dst_nodata is not None.
+    dst_transform: affine transform object, optional
+        Target affine transformation.  Required if source and destination
+        are ndarrays.  Will be derived from target if it is a rasterio Band.
+    dst_crs: dict, optional
+        Target coordinate reference system.  Required if source and destination
+        are ndarrays.  Will be derived from target if it is a rasterio Band.
+    dst_nodata: int or float, optional
+        The nodata value used to initialize the destination; it will remain
+        in all areas not covered by the reprojected source.  Defaults to the
+        nodata value of the destination image (if set), the value of
+        src_nodata, or 0 (gdal default).
```

Comment:
No, you're correct @brendan-ward. I just had a brain malfunction from rushing through too many issues.